python/ray/tests/test_tracing.py

```diff
@@ -0,0 +1,187 @@
+import asyncio
+import glob
+import json
+import os
+import pytest
+import shutil
+import tempfile
+
+import ray
+from ray.test_utils import check_call_ray
+
+# Create temporary spans folder for trace output.
+spans_dir = tempfile.gettempdir() + "/spans"
+
+
+@pytest.fixture()
+def cleanup_dirs():
+    """Cleanup temporary spans_dir folder at beginning and end of test."""
+    if os.path.exists(spans_dir):
+        shutil.rmtree(spans_dir)
+    os.makedirs(spans_dir)
+    yield
+    # Enable tracing only sets up tracing once per driver process.
+    # We set ray.__traced__ to False here so that each
+    # test will re-set up tracing.
+    ray.__traced__ = False
+    if os.path.exists(spans_dir):
+        shutil.rmtree(spans_dir)
+
+
+@pytest.fixture()
+def ray_start_cli_tracing(scope="function"):
+    """Start ray with tracing-startup-hook, and clean up at end of test."""
+    check_call_ray([
+        "start", "--head", "--tracing-startup-hook",
+        "ray.tests.enable_tracing:_setup_tracing"
+    ], )
+    ray.init(address="auto")
+    yield
+    ray.shutdown()
+    check_call_ray(["stop", "--force"])
+
+
+@pytest.fixture()
+def ray_start_init_tracing(scope="function"):
+    """Call ray.init with tracing-startup-hook, and clean up at end of test."""
+    ray.init(_tracing_startup_hook="ray.tests.enable_tracing:_setup_tracing")
```

Comment:
```suggestion
    ray.init(_tracing_startup_hook="ray.tests.enable_tracing:_setup_tracing")
```
I find this colon syntax a little weird -- is it a standard from somewhere else? Otherwise I'd say we should just do `ray.tests.enable_tracing._setup_tracing`

Reply:
this is using the uvicorn import_from_string function - https://github.com/encode/uvicorn/blob/master/uvicorn/importer.py

Reply:
The colon syntax is pretty common for Python especially in the web world:
`gunicorn -w 4 myapp:app`
`uvicorn my_module:app_obj`